addons/base_vat/models/res_partner.py

```diff
@@ -171,6 +171,13 @@ def _construct_constraint_msg(self, country_code):
 
     __check_vat_ch_re = re.compile(r'E([0-9]{9}|-[0-9]{3}\.[0-9]{3}\.[0-9]{3})(MWST|TVA|IVA)$')
 
+    def check_vat_xi(self, vat):
```

Comment:
Just last comment: try to keep the functions `check_vat_XX` ordered alphabetically. You are also cutting two definitions tightly linked: `__check_vat_ch_re` and `check_vat_ch`

Reply:
done